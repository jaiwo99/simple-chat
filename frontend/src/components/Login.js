import React from 'react';

export default class Login extends React.Component {

  constructor(props) {
    super(props);

    this.store = this.props.store;
    this.reload = this.props.reloadFn;

    this.state = {
      author: ''
    };
  }

  render() {
    return (<div className="login">
        <form onSubmit={this.login.bind(this)}>
          <input type="text" value={this.state.author} onChange={this.handleChange.bind(this)}/>
          <button type="submit">Enter</button>
        </form>
      </div>
    );
  }

  login(e) {
    this.store.save(this.state.author);
    this.reload();

    e.preventDefault();
  }

  handleChange(e) {
    this.setState({
      author: e.target.value
    });
  }
}
