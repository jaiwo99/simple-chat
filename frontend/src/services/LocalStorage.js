const KEY = 'CHAT_AUTHOR';

export default class LocalStorage {

  save(name) {
    localStorage.setItem(KEY, name);
  }

  read() {
    return localStorage.getItem(KEY);
  }
}
