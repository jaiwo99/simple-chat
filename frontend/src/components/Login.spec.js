import React from "react";
import LocalStorage from "../services/LocalStorage";
import renderer from "react-test-renderer";
import sinon from "sinon";
import Login from "./Login";

describe('Login Component', () => {

  test('should submit username', () => {
    localStorage.clear();

    const spy = sinon.spy();
    const store = new LocalStorage();
    const component = renderer.create(
      <Login store={store} reloadFn={spy}/>
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    expect(localStorage.getItem('CHAT_AUTHOR')).toBeNull();

    tree.children[0].children[0].props.onChange({
      target: {
        value: 'max'
      },
      preventDefault: () => {
      }
    });

    tree.children[0].props.onSubmit({
      preventDefault: () => {
      }
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(localStorage.getItem('CHAT_AUTHOR')).toBe('max');
    expect(spy.called).toBe(true);
  });
});

