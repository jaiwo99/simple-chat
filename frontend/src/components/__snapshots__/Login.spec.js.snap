// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should submit username 1`] = `
<div
  className="login"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Enter
    </button>
  </form>
</div>
`;

exports[`Login Component should submit username 2`] = `
<div
  className="login"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value="max"
    />
    <button
      type="submit"
    >
      Enter
    </button>
  </form>
</div>
`;
